quinn-udp/src/unix.rs

```diff
@@ -383,31 +383,33 @@ fn send(state: &UdpSocketState, io: SockRef<'_>, transmit: &Transmit<'_>) -> io:
         hdrs[i].msg_datalen = chunk.len();
         cnt += 1;
     }
-    let n = unsafe { sendmsg_x(io.as_raw_fd(), hdrs.as_ptr(), cnt as u32, 0) };
-    if n >= 0 {
-        return Ok(());
-    }
-    let e = io::Error::last_os_error();
-    match e.kind() {
-        io::ErrorKind::Interrupted => {
-            // Retry the transmission
-        }
-        io::ErrorKind::WouldBlock => return Err(e),
-        _ => {
-            // Other errors are ignored, since they will usually be handled
-            // by higher level retransmits and timeouts.
-            // - PermissionDenied errors have been observed due to iptable rules.
-            //   Those are not fatal errors, since the
-            //   configuration can be dynamically changed.
-            // - Destination unreachable errors have been observed for other
-            // - EMSGSIZE is expected for MTU probes. Future work might be able to avoid
-            //   these by automatically clamping the MTUD upper bound to the interface MTU.
-            if e.raw_os_error() != Some(libc::EMSGSIZE) {
-                log_sendmsg_error(&state.last_send_error, e, transmit);
+    loop {
+        let n = unsafe { sendmsg_x(io.as_raw_fd(), hdrs.as_ptr(), cnt as u32, 0) };
+        if n == -1 {
```

Comment:
For both of these, how do we feel about inverting this condition to reduce rightward drift at the cost of duplicating the `return Ok(());`?

And/or have you considered deduplicating more of this logic? These `match` expressions are pretty much the same...

Reply:
The entire loop could probably be extracted into a function with the syscall being an `FnMut` parameter?

Reply:
We can of have the same loop three times actually.

Reply:
I can provide another patch de-duplicating the loop logic. That said, I suggest doing that in a separate pull request.

@djc @thomaseizinger fine by you?

Reply:
Works for me.